applications/asset_tracker/src/main.c

```diff
@@ -567,7 +567,7 @@ static void cloud_connect_work_fn(struct k_work *work)
 	/* Attempt cloud connection */
 	ret = cloud_connect(cloud_backend);
 	if (ret != CLOUD_CONNECT_RES_SUCCESS) {
-		k_delayed_work_cancel(&cloud_reboot_work);
+		k_work_cancel_delayable(&cloud_reboot_work);
```

Comment:
There are lots of these related to _cloud_reboot_work_ . Are there other consequences than an extra reboot if the cancel fails?

Reply:
@jayteemo I think you have most knowledge about the reboot mechanisms in the asset tracker.

Reply:
Well in general the cancelation can fail if the canceled work is already being processed by the work thread. I don't see a way to prevent the work from being executed at this point (other than aborting the thread, which is not what we want I guess).

Reply:
Btw, if you're concerned about syncrhnization issues, when the cancelled work is already being exectured, there's `k_work_cancel_delayable_sync()` which should block until the work is finished (in case it's already being handled by the work thread).

Reply:
> @jayteemo I think you have most knowledge about the reboot mechanisms in the asset tracker.

The reboot work is in asset tracker is just used as a fail safe for connect/disconnect problems.

Reply:
keeping as is